### PR TITLE
User menu

### PR DESCRIPTION
**Pull request by boutell**
## Summary

* See changelog

## What are the specific steps to test this change?

You could temporarily change the new code in `login/index.js` that adds the "Log Out" option via this feature to not add it at all, or add an additional item (that emits an event that currently does nothing). etc. This would prove we're not using the old hard-coded menu anymore.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarMenu.vue [hunk 1]
```diff
@@ -54,10 +54,11 @@
         <Component
           v-if="item.options.component"
           :is="item.options.component"
-          :key="item.name"
+          :key="`${item.name}.component`"
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarMenu.vue:57]: Unrelated linter fix.

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarMenu.vue [hunk 2]
```diff
@@ -54,10 +54,11 @@
...
         />
         <AposButton
-          v-else :key="item.name"
+          v-else
+          :key="`${item.name}.fallback`"
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarMenu.vue:61]: Unrelated linter fix.

File: modules/@apostrophecms/login/index.js [hunk 3]
```diff
@@ -850,7 +851,20 @@ module.exports = {
           namespace,
           key: username
         });
+      },
+
+      addToAdminBar() {
```
Comment (boutell) [inline modules/@apostrophecms/login/index.js:856]: How the log out item gets added now. A good method to hack on for testing.

File: modules/@apostrophecms/login/index.js [hunk 4]
```diff
@@ -850,7 +851,20 @@ module.exports = {
...
+        self.apos.adminBar.add(
+          `${self.__meta.name}-logout`,
+          'apostrophe:logOut',
+          false,
+          {
+            user: true,
+            last: true
```
Comment (boutell) [inline modules/@apostrophecms/login/index.js:863]: So "log out" winds up last, which feels like the right UX.

**Review by gregvanbrug — APPROVED** [review]
